### PR TITLE
Add io.unobserved.espansoGUI

### PR DESCRIPTION
**Pull request by rickykresslein**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
    - My app requires access to the xdg-config/espanso directory to edit espanso files (this is the reason the app exists)
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **I am the creator of the app**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission

### CODE REVIEW

File: io.unobserved.espansogui.json [hunk 1]
```diff
@@ -0,0 +1,39 @@
+{
+  "app-id": "io.unobserved.espansoGUI",
+  "runtime": "org.freedesktop.Platform",
+  "runtime-version": "22.08",
```
Comment (hfiguiere) [inline io.unobserved.espansogui.json:4]: 23.08 is the current runtime

File: io.unobserved.espansogui.json [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    "--share=ipc",
+    "--socket=fallback-x11",
+    "--device=dri",
+    "--socket=wayland",
+    "--filesystem=xdg-config/espanso"
+  ],
+  "build-options": {
+    "append-path": "/usr/lib/sdk/rust-stable/bin",
+    "build-args" : [
+      "--share=network"
```
Comment (hfiguiere) [inline io.unobserved.espansogui.json:20]: build must be offline. This is not allowed.

File: io.unobserved.espansogui.json [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  "sdk-extensions": [
+    "org.freedesktop.Sdk.Extension.rust-stable"
+  ],
+  "command": "espanso-gui",
+  "finish-args": [
+    "--share=ipc",
+    "--socket=fallback-x11",
+    "--device=dri",
+    "--socket=wayland",
+    "--filesystem=xdg-config/espanso"
```
Comment (hfiguiere) [inline io.unobserved.espansogui.json:15]: why not shipping espanso here instead?

File: io.unobserved.espansogui.json (hunk 2 above)
Comment (hfiguiere) [reply io.unobserved.espansogui.json:20]: see https://github.com/flatpak/flatpak-builder-tools/tree/master/cargo

File: io.unobserved.espansogui.json (hunk 1 above)
Comment (rickykresslein) [reply io.unobserved.espansogui.json:4]: I can't remember exactly which package it was, but when trying to build with 23.08 one of the packages required is still only available as a nightly and doesn't have a stable version. Therefore, I went with 22.08 and I will upgrade when there is a stable version available.

File: io.unobserved.espansogui.json (hunk 2 above)
Comment (rickykresslein) [reply io.unobserved.espansogui.json:20]: Sorry, this was left over from my tests and I forgot to remove it. I will do that now.

File: io.unobserved.espansogui.json (hunk 3 above)
Comment (rickykresslein) [reply io.unobserved.espansogui.json:15]: Sorry, not sure what you mean. If you mean packaging espanso to put on Flathub, I am not the maintainer and espanso requires permissions that may be beyond the limits of Flatpak.

File: .github/pull_request_template.md
Comment (hfiguiere) [inline .github/pull_request_template.md:1]: please remove the whole `.github` directory. It shouldn't be here.

File: CONTRIBUTING.md
Comment (hfiguiere) [inline CONTRIBUTING.md:1]: remove this file

File: COPYING
Comment (hfiguiere) [inline COPYING:1]: remove this file

File: README.md
Comment (hfiguiere) [inline README.md:1]: remove this file

File: io.unobserved.espansogui.json (hunk 1 above)
Comment (hfiguiere) [reply io.unobserved.espansogui.json:4]: rust-stable IS available for 23.08

**Comment by hfiguiere** [comment]
bot, build io.unobserved.espansoGUI

File: io.unobserved.espansogui.json
Comment (hfiguiere) [inline io.unobserved.espansogui.json:1]: the file name doen't match the app-id

**Comment by rickykresslein** [comment]
@hfiguiere, sorry, somehow the main branch must have gotten merged into this branch. Everything you mentioned should be fixed now. Thank you for taking the time to go over this.

**Comment by hfiguiere** [comment]
bot, build io.unobserved.espansoGUI

**Comment by hfiguiere** [comment]
the appid in the appstream file is incorrect.

**Comment by rickykresslein** [comment]
bot, build io.unobserved.espansoGUI

**Comment by rickykresslein** [comment]
The build failed due to a missing 128x128 icon, but according to the Flathub requirements:

> Applications must provide application icons in at least 128×128px size, or a scalable SVG. Application icons should either be included as scalable SVGs or high-resolution PNGs, and unless they are directly copied from upstream during the package building, they should be limited to the minimum number.

I have a scalable SVG included, and it works in my own Flatpak build test (the icon is visible for the app in Gnome) so I'm not sure what I'm missing here. Do I need to include another 128x128 png?

**Comment by bbhtt** [comment]
> The build failed due to a missing 128x128 icon, but according to the Flathub requirements:
> 
> > Applications must provide application icons in at least 128×128px size, or a scalable SVG. Application icons should either be included as scalable SVGs or high-resolution PNGs, and unless they are directly copied from upstream during the package building, they should be limited to the minimum number.
> 
> I have a scalable SVG included, and it works in my own Flatpak build test (the icon is visible for the app in Gnome) so I'm not sure what I'm missing here. Do I need to include another 128x128 png?

1. Wrong launchable id in the appdata
2. You're installing it to the wrong location, it's an appdata.xml it goes into `.../share/appdata/{id}.appdata.xml`, metainfo is the newer format which goes into `.../share/metainfo/{id}.metainfo.xml`

**Comment by rickykresslein** [comment]
bot, build io.unobserved.espansoGUI

File: io.unobserved.espansoGUI.json [hunk 4]
```diff
@@ -28,7 +28,7 @@
         {
           "type": "archive",
           "url": "https://github.com/unobserved-io/espanso-gui/releases/download/2023.09.15/espanso-gui-2023.09.15.tar.xz",
-          "sha256": "82376cddfd58d02fa3fbe5321e95ab0c7d5d8829c7c6ba09ca9d27926e4e34b6"
+          "sha256": "f34a773d6ee604a13915c7c2bec22994cdbbbe1c40dfa21c850080923479a236"
```
Comment (hfiguiere) [inline io.unobserved.espansoGUI.json:31]: "retagging" is a bad practice.

Comment (rickykresslein) [reply io.unobserved.espansoGUI.json:31]: Sorry, I'm not sure what you mean, could you explain?

**Comment by rickykresslein** [comment]
Well that was a journey 😅 Thank you @hfiguiere and @bbhtt for your patience. Hopefully it's good to go now.

File: io.unobserved.espansoGUI.json (hunk 4 above)
Comment (hfiguiere) [reply io.unobserved.espansoGUI.json:31]: you regenerated a tarball with the same name. That's retagging. It's a bad practice (nothing to do with Flathub). It should be a new release.

Comment (rickykresslein) [reply io.unobserved.espansoGUI.json:31]: Oh, okay, thank you.

**Pull request closed without merging**